csv/parse_stream_test.ts

```diff
@@ -48,12 +42,11 @@ Deno.test({
     const reader = readable.getReader();
     assertEquals(await reader.read(), { done: false, value: ["id", "name"] });
     assertEquals(await reader.read(), { done: false, value: ["1", "foo"] });
-    const error = await assertRejects(() => reader.read());
-    assert(error instanceof ParseError);
-    assertEquals(error.startLine, 4);
-    assertEquals(error.line, 5);
-    assertEquals(error.column, 0);
```

Comment:
These information will be unavailable. Is that fine?

WDYT? cc @iuioiua

Reply:
Ah, true. We can store this data within [`Error.cause`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause).

Reply:
> We can store this data within [Error.cause](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause).

Sounds a reasonable alternative to me. That use case is recommended in the 2nd example https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause#providing_structured_data_as_the_error_cause

But I'm a bit concerned about the unavailability of the type. `err.cause` seems always `unknown` type. The user need to check type and properties of `cause` before accessing it. On the other hand, if it's kept `ParseError`, these properties can be accessed after the user checked `err instanceof ParseError`, which feels a bit easier and clearer to me. What do you think?

Reply:
Good point. Actually, now, I'm wondering whether it's worth having those extra details at all. `JSON.parse` doesn't do this. E.g. `JSON.parse("[1,2,3,")` throws an error with `.cause` undefined. We might be fine without. I suggest removing these properties. If it's a sufficiently significant issue in the future, we can revisit. WDYT?

Reply:
Fair enough. Let's try this and see the community reactions.